Implement category, filter UI

arambaghdasintellias opened the pull request.

Commit:
Implement category UI

Commit:
Implement filter

Commit:
Implement filter

Commit:
Fix issues

Commit:
Set home, work addresses

Commit:
Show available services in the parking

Commit:
Refactor filter functionality

Commit:
Reset user settings

Commit:
Show suggestions

Commit:
Fix location values

jj123640131 commented on app/src/main/java/telenav/demo/app/utils/CategoryAndFiltersUtil.kt:
```diff
@@ -72,6 +185,14 @@ object CategoryAndFiltersUtil {
+                fourthStar.setImageDrawable(context.getDrawable(R.drawable.ic_star_empty))
+                fifthStar.setImageDrawable(context.getDrawable(R.drawable.ic_star_empty))
+            }
+
```
Should hide these stars when rating value is 0.0

arambaghdasintellias replied:
It hides stars in case of rating value is 0.0 

            0.0 -> {
                firstStar.setImageDrawable(context.getDrawable(R.drawable.ic_star_empty))
                secondStar.setImageDrawable(context.getDrawable(R.drawable.ic_star_empty))
                thirdStar.setImageDrawable(context.getDrawable(R.drawable.ic_star_empty))
                fourthStar.setImageDrawable(context.getDrawable(R.drawable.ic_star_empty))
                fifthStar.setImageDrawable(context.getDrawable(R.drawable.ic_star_empty))
            }

jj123640131 commented on app/src/main/java/telenav/demo/app/map/MapActivity.kt:
```diff
@@ -267,25 +303,133 @@ class MapActivity : AppCompatActivity() {
+
+    var searchListBottomFragment: SearchListBottomFragment? = null
+    fun showSearchListBottomFragment() {
+        searchListBottomFragment = SearchListBottomFragment.newInstance(hotCategoryTag)
```
Fragment will appear multiple times after multiple clicks, there should be a logic to prevent that.

jj123640131 commented on app/src/main/java/telenav/demo/app/map/MapActivity.kt:
```diff
@@ -267,25 +303,133 @@ class MapActivity : AppCompatActivity() {
+        entity_details.visibility = View.VISIBLE
+        top_navigation_panel.visibility = View.VISIBLE
+
+        navigation_header.text = hotCategoryName
```
The text of navigation_header should be what user inputs first.

arambaghdasintellias replied:
User can select from recent search list, favorite list without typing text, what should show in that case ?

jj123640131 commented on app/src/main/java/telenav/demo/app/map/MapActivity.kt:
```diff
@@ -267,25 +303,133 @@ class MapActivity : AppCompatActivity() {
     }
+
+    fun showEntityDetails(searchResult: SearchResult, entity: Entity) {
+        entity_details.visibility = View.VISIBLE
```
There should be a logic to draw pin on the map according to the category in searchResult.

arambaghdasintellias replied:
It happens in the addSearchResultsOnMap function in MapFragment.

jj123640131 commented on app/src/main/java/telenav/demo/app/map/MapFragment.kt:
```diff
@@ -109,65 +76,90 @@ class MapFragment : Fragment(), GoogleMap.OnInfoWindowClickListener,
+        currentSearchHotTag: String?,
+        shouldOpenEntityDetails: Boolean = false) {
+
+        if (entity.place == null) {
```
There are two types in Entity, refer to EntityType, please refine the logic to display entity on the map.

jj123640131 commented on app/src/main/java/telenav/demo/app/map/MapFragment.kt:
```diff
@@ -109,65 +76,90 @@ class MapFragment : Fragment(), GoogleMap.OnInfoWindowClickListener,
+        entity: Entity,
+        currentSearchHotTag: String?,
+        shouldOpenEntityDetails: Boolean = false) {
+
```
Seems the click event for makers is missing.

arambaghdasintellias replied:
It happens in this function in MapFragment
override fun onMarkerClick(marker: Marker?): Boolean

jj123640131 commented on app/src/main/java/telenav/demo/app/map/MapFragment.kt:
```diff
@@ -109,65 +76,90 @@ class MapFragment : Fragment(), GoogleMap.OnInfoWindowClickListener,
-        customInfoWindowAdapter.setSearchResult(searchResultList[marker?.zIndex!!.toInt() - 1])
-        marker.showInfoWindow()
+        val index= marker?.zIndex!!.toInt() - 1
+        val searchResult = searchResultList[index]
```
Please add validation check for 'index'.

arambaghdasintellias replied:
Done

jj123640131 commented on app/src/main/java/telenav/demo/app/search/EntityDetailsFragment.kt:
```diff
@@ -0,0 +1,158 @@
+
+        binding?.entityName?.text = searchResult?.name
+
+        if (searchResult?.phoneNo.isNullOrEmpty()) {
```
Please display correct rating, open time and open state.
for rating, refer to Entity.Facets.rating
for open time, refer to Entity.FacetOpenHours.regularOpenHours
for open state, refer to Entity.FacetOpenHours.openNow

arambaghdasintellias replied:
Fixed.

Commit:
Show more categories

Commit:
Fix issues in the word prediction

Commit:
Validate marker index

Commit:
Implement more categories request

Commit:
Apply parking time filter

Commit:
Collapse entity details view on map click

Commit:
Fix bottom sheet state

Commit:
Show keyboard on search input

Commit:
Fix UI issues

Commit:
Fix UI issues

Commit:
Fix UI issues in search list

Commit:
Fix UI issues in search bottom view

Commit:
Fix UI issues in recent data view

Commit:
Fix UI issues

Commit:
Code cleanup

Commit:
Fix entity UI issues

Commit:
Fix UI issues

Commit:
Fix UI issues in entity details

Commit:
Show yelp reviews

Commit:
Handle system back action

Commit:
Update user info icons

Commit:
Fix UI issues

Commit:
bug fix.

VegeDoggie committed:
Hot Category Icons replacement.

VegeDoggie committed:
Use RecyclerView to implement CategoryView instead of Flow view in MapActivity.

VegeDoggie committed:
refactor sub category view with recyclerView.

VegeDoggie committed:
Remove useless class and files.

VegeDoggie committed:
Adjust icon position in the Pin marker.

VegeDoggie committed:
Use MaterialButton to center the icon and text in the button.

VegeDoggie committed:
Adjust navigation panel's margin to align with map's button.

VegeDoggie committed:
Update initialization activity to new designed UI, use Icon instead of text in splash screen.

VegeDoggie committed:
Hide keyBoard while dragging search list page, avoid keyboard stay visible after getting back to map activity.

VegeDoggie committed:
Hide review text when collapsed.

VegeDoggie committed:
Adjust EV charger filter page's layout.

VegeDoggie committed:
Add new UI elements to settings, including:
1. server list spinner
2. cvp location editText
3. search area location editText

VegeDoggie committed:
UI modification on settings page.

VegeDoggie committed:
Remove useless elements and code.

VegeDoggie committed:
Enable cvp locations in setting page, basic functions ready.
But, data still contained in settings, not delivered to MapActivity yet.

VegeDoggie committed:
Make Toast when latLong is invalid.
Finish input when Enter is entered.

VegeDoggie committed:
Enable CVP location and search area location when searching.

VegeDoggie committed:
Adjust category sequence to avoid text overlap.

VegeDoggie committed:
Move map camera to CVP location instead of gps location.

VegeDoggie committed:
Index out of bounds protection.

VegeDoggie committed:
New string array placeholder for server list.

VegeDoggie committed:
Server list feature.
1. [Mock]Read server list json from file.
2. Init sdks using selected server.
3. Support selecting servers from settings page, with spinner widget.
4. Changing server will cause App restart when leaving settings page.
5. Exiting MapActivity will kill process to make sure SDK Initialization properly.

VegeDoggie committed:
Upgrade minSdkVersion to 26, import BuildConfig fields to store rsa keys.

VegeDoggie committed:
Support fetch server list from web server:
1. use server list from web server instead of assets.
2. Add rsaDecipher to decrypt apiKeys and apiSecrets.
3. Json structure adjustment, removing "products" field, move every fields in it outside.

VegeDoggie committed:
Refactor settingsActivity, use viewModel to control data and drive views.
Currently server list moved into vm, rest of the view data need more work to refactor.

VegeDoggie committed:
Move location settings data into viewModel.

VegeDoggie committed:
Add Quick Links feature in location settings.

VegeDoggie committed:
Integrated Instabug to App.
Hide select index data dir button due to some permission conflict, hardcode dir to "filesDir/indexData"

YesIamJeremyChen committed:
Fix the bug that the ev filter is not working well

VegeDoggie committed:
Using cvp location (if exist) to generate home area.

VegeDoggie committed:
Fix ev filters bug.

VegeDoggie committed:
Smart Region & Server selection (Show server list options according by current search location).

VegeDoggie committed:
New DateTimePicker for parking filter params, with smooth collapse/expand animation.

VegeDoggie committed:
Add destination prediction feature to app, request prediction from server when started, display result in recommend list.

VegeDoggie committed:
Add User ID settings, allow user to change id by typing.
User ID will be used to request predict location.

VegeDoggie committed:
Add bounce animation to splash screen.

VegeDoggie committed:
UI tuning:
1. Search bar layout adjustment;
2. Settings spinner text color;
3. RoundCorners for handle bar at home page;
4. Disable search area location's widget after enable "follow CVP".

VegeDoggie committed:
Show marker of predict location on map, fill detail page with address information only.

VegeDoggie committed:
Extract all handle bar view to one View xml file, once modify, effect all views.

VegeDoggie committed:
Fix entity detail page and settings page's button issue.

VegeDoggie committed:
Adjust general filters layout.

Commit:
add keystore files

Commit:
Init jenkins build for telenav

jj123640131 committed:
Update gradle.properties

Commit:
fix conflict

Commit:
add artifactId

Commit:
explore feature.

Commit:
explore feature.

Commit:
explore feature.

Commit:
explore feature.
